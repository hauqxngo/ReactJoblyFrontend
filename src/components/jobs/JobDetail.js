import React, { useState, useEffect, useContext } from "react";
import { Button } from "reactstrap";
import UserContext from "../../UserContext";
import { Card, CardBody, CardText, CardTitle } from "reactstrap";

const JobDetail = ({ id, title, salary, equity }) => {
  const [applied, setApplied] = useState();
  const { hasApplied, apply } = useContext(UserContext);

  useEffect(
    (updateAppliedButton) => {
      setApplied(hasApplied(id));
    },
    [id, hasApplied]
  );

  // Apply for a job
  async function handleApply() {
    if (hasApplied(id)) return;
    apply(id);
    setApplied(true);
  }

  return (
    <div>
      <Card body color="secondary mx-5 my-2" outline>
        <CardBody>
          <CardTitle className="text-success" tag="h5">
            {title}
          </CardTitle>
          <CardText className="text-dark">
            <p>Salary: {salary === null ? "Negotiable" : `$${salary}`}</p>
            <p>Equity: {equity === null ? "N/A" : equity}</p>
          </CardText>
          <div class="d-grid gap-2 d-md-flex justify-content-md-end">
            <Button
              type="submit"
              color="success me-md-2"
              onClick={handleApply}
              disabled={applied}
            >
              {applied ? "APPLIED" : "APPLY"}
            </Button>
          </div>
        </CardBody>
      </Card>
    </div>
  );
};

export default JobDetail;
