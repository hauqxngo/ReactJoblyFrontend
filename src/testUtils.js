import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  first_name: "test",
  last_name: "user",
  email: "test@test.com",
  photo_url: null,
};

const UserProvider = ({
  children,
  currentUser = demoUser,
  hasApplied = () => false,
}) => (
  <UserContext.Provider value={{ currentUser, hasApplied }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
