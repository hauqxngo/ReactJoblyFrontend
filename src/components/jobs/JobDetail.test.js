import React from "react";
import { render } from "@testing-library/react";
import JobDetail from "./JobDetail";
import { UserProvider } from "../../testUtils";

it("matches snapshot", () => {
  let job = { title: "Software Engineer", salary: 1000000, equity: 10 };
  const { asFragment } = render(
    <UserProvider>
      <JobDetail job={job} />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
