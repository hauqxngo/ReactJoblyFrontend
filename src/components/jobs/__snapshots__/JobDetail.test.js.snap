// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="card card-body border-secondary mx-5 my-2"
    >
      <div
        class="card-body"
      >
        <h5
          class="text-success card-title"
        />
        <p
          class="text-dark card-text"
        />
        <p>
          Salary: $undefined
        </p>
        <p>
          Equity: 
        </p>
        <p />
        <div
          class="d-grid gap-2 d-md-flex justify-content-md-end"
        >
          <button
            class="btn btn-success me-md-2"
            type="submit"
          >
            APPLY
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
